src/Libraries/Microsoft.Extensions.AI.Abstractions/SpeechToText/SpeechToTextResponse.cs

```diff
@@ -0,0 +1,118 @@
+
+    /// <summary>Initializes a new instance of the <see cref="SpeechToTextResponse"/> class.</summary>
+    [JsonConstructor]
+    public SpeechToTextResponse()
+    {
+    }
+
+    /// <summary>Initializes a new instance of the <see cref="SpeechToTextResponse"/> class.</summary>
+    /// <param name="contents">The contents for this response.</param>
+    public SpeechToTextResponse(IList<AIContent> contents)
+    {
+        _contents = Throw.IfNull(contents);
+    }
+
+    /// <summary>Initializes a new instance of the <see cref="SpeechToTextResponse"/> class.</summary>
+    /// <param name="content">Content of the response.</param>
+    public SpeechToTextResponse(string? content)
+        : this(content is null ? [] : [new TextContent(content)])
+    {
+    }
+
+    /// <summary>Gets or sets the start time of the text segment in relation to the full audio speech length.</summary>
+    public TimeSpan? StartTime { get; set; }
+
+    /// <summary>Gets or sets the end time of the text segment in relation to the full audio speech length.</summary>
+    public TimeSpan? EndTime { get; set; }
+
+    /// <summary>Gets or sets the ID of the speech to text response.</summary>
+    public string? ResponseId { get; set; }
+
+    /// <summary>Gets or sets the model ID used in the creation of the speech to text completion.</summary>
+    public string? ModelId { get; set; }
+
+    /// <summary>Gets or sets the raw representation of the speech to text completion from an underlying implementation.</summary>
+    /// <remarks>
+    /// If a <see cref="SpeechToTextResponse"/> is created to represent some underlying object from another object
+    /// model, this property can be used to store that original object. This can be useful for debugging or
+    /// for enabling a consumer to access the underlying object model if needed.
+    /// </remarks>
+    [JsonIgnore]
+    public object? RawRepresentation { get; set; }
+
+    /// <summary>Gets or sets any additional properties associated with the speech to text completion.</summary>
+    public AdditionalPropertiesDictionary? AdditionalProperties { get; set; }
+
+    /// <summary>
+    /// Gets or sets the text of the first <see cref="TextContent"/> instance in <see cref="Contents" />.
+    /// </summary>
+    /// <remarks>
+    /// If there is no <see cref="TextContent"/> instance in <see cref="Contents" />, then the getter returns <see langword="null" />,
+    /// and the setter adds a new <see cref="TextContent"/> instance with the provided value.
+    /// </remarks>
+    [JsonIgnore]
+    public string? Text
+    {
+        get => Contents.OfType<TextContent>().FirstOrDefault()?.Text;
+        set
+        {
+            if (Contents.OfType<TextContent>().FirstOrDefault() is { } textContent)
+            {
+                textContent.Text = value;
+            }
+            else if (value is not null)
+            {
+                Contents.Add(new TextContent(value));
+            }
+        }
+    }
```

Comment:
This should mirror the behavior we now have for ChatResponse.Text, which concatenates the text of all messages. It also can't be null anymore. And has no setter.